docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,304 @@
+
+`Equal()` can also be seen as redundant if we set `Hash()` in the right way that we can just compare evidence hashes for uniqueness. We already use hashes for database lookups
+
+`Type()` could be a new method which would be helpful in constructing abci evidence
+
+`ToABCI()` is another alternative that was written as a TODO in the actual code base.
+
+
+## The State of Amnesia
+
+TBC
+
+## Censorship
+
+A second problem brought up in a prior [issue](https://github.com/tendermint/tendermint/issues/4593) is the notion of evidence censorship. Consider the following scenario: I have greater than 1/3 validator power. I perform an attack either on a light client of on the main network that was detected. Whenever someone proposes a block that contains evidence I abstain from voting meaning that it never is committed. When it is my turn to propose, I propose a block that does not include the evidence, everyone else votes and that block gets committed. Eventually the chain reaches a height and time where the evidence has expired and any attempt to punish my attack has been thwarted.
+
+This would most likely need an RFC (in order to explore possible solutions). As far as I can see there is a 0% approach a 50% approach and 100% approach in terms of the level of intervention.
+
+The 0% approach would just be simply to leave this as is.
+
+The 50% approach would be that validators would consider a proposal invalid if it didn't contain at least one of the evidence that that validator had and thus wouldn't vote on it. This solution is far less intrusive to the current consensus algorithm but it also doesn't quite solve the problem of censorship, rather this protocol will keep the chain halted until off-chain governance can intervene.
+
+The 100% approach is a lot more intricate and would be to find a protocol where upon receiving a valid proposal with evidence, current validators that had been shown to have misbehaved would be barred from voting such that the total voting power would be reduced by the voting power of the malicious validators and this would allow for 2/3 consensus to be reached regardless of what the malicious validators did.
+
+
+## Proposal
+
+A Hybrid design
+
+Evidence has the following simple interface:
+
+```go
+type Evidence interface {  //proposed
+  Height() int64                                     // height of the equivocation
+  Bytes() []byte                                     // bytes which comprise the evidence
+  Hash() []byte                                      // hash of the evidence
+  ValidateBasic() error
+  String() string
+}
+```
+
+We have two recognized evidence that complies with it
```

Comment:
"There are two concrete types that fulfill the `Evidence` interface." 

(Is that the same thing that you mean?)

Reply:
yup